api/src/main/java/org/openmrs/module/queue/model/QueueEntryTransition.java

```diff
@@ -49,8 +55,8 @@ public QueueEntry constructNewQueueEntry() {
 		    newPriorityComment == null ? queueEntryToTransition.getPriorityComment() : newPriorityComment);
 		queueEntry.setStatus(newStatus == null ? queueEntryToTransition.getStatus() : newStatus);
 		queueEntry.setSortWeight(queueEntryToTransition.getSortWeight());
-		queueEntry.setLocationWaitingFor(queueEntryToTransition.getLocationWaitingFor());
-		queueEntry.setProviderWaitingFor(queueEntryToTransition.getProviderWaitingFor());
+		queueEntry.setLocationWaitingFor(newLocationWaitingFor);
+		queueEntry.setProviderWaitingFor(newProviderWaitingFor);
```

Comment:
It seems a little inconsistent here to do this vs. what we do for status, priority, etc.  For the existing fields, we handle the logic here for setting them to existing fields.  For these new properties, we handle that logic in the REST controller.  Would it make sense to be consistent?